esp/esp/program/views.py

```diff
@@ -536,13 +536,11 @@ def newprogram(request):
                 new_scrmi.save()
                 # Copy tags from old program
                 ct = ContentType.objects.get_for_model(old_prog)
-                tags = Tag.objects.filter(content_type=ct, object_id=old_prog.id)
-                for tag in tags:
-                    tag.pk = None
-                    tag.id = None
-                    tag.target = new_prog
-                    tag.object_id = new_prog.id
-                    tag.save()
+                old_tags = Tag.objects.filter(content_type=ct, object_id=old_prog.id)
+                for old_tag in old_tags:
+                    new_tag, _ = Tag.objects.get_or_create(key=old_tag.key, content_type=ct, object_id=new_prog.id)
```

Comment:
I now don't get an error, but I do think there's a bit of undesired behavior. The past program tag for sibling discount is overriding the tag created from the form for creating the new program, which means I enter a new value for the sibling discount in the program creation form but then it's really being set to the past value without telling me. I think instead of the "get_or_create", the code should check if that tag has already been created and only copy the old program's tag if it doesnt exist?  Not sure if this would negatively impact tags other than sibling discount though.

Reply:
OK, how about now @kkbrum?